### PR TITLE
HFX-1717: Backport of the TLS 1.2 support to XS 6.1

### PR DESCRIPTION
**Pull request by fillzero**

### CODE REVIEW

**Comment by fillzero** [comment]
hi @thomassa ,
Please help to merge these commits to tampa-bugfix branch.
Thank you.

**Comment by thomassa** [comment]
Are these changes present already on any other -bugfix or -lcm branch?

In this pull-request, there are two commits that are immediately followed by commits that revert them.
Those four commits should be removed (unless you know some special reason to keep them).

**Comment by fillzero** [comment]
Hi @thomassa ,
Thank you for your comments, I removed those four commits.

File: ocaml/xapi/cli_operations.ml [hunk 1]
```diff
@@ -3956,13 +3972,6 @@ let host_reset_cpu_features printer rpc session_id params =
 			get_host_from_session rpc session_id in
 	Client.Host.reset_cpu_features rpc session_id host
 
-let host_set_ssl_legacy printer rpc session_id params =
```
Comment (thomassa) [inline ocaml/xapi/cli_operations.ml]: Why are you removing this CLI command?

File: ocaml/xapi/cli_frontend.ml [hunk 2]
```diff
@@ -783,14 +783,24 @@ let rec cmdtable_data : (string*cmd_spec) list =
       flags=[];
     };
 
-   "host-enable-display",
```
Comment (thomassa) [inline ocaml/xapi/cli_frontend.ml]: Why are you removing host-enable-display?

**Comment by thomassa** [comment]
Good: that helps.

Have the remaining changes had a thorough review in pull-requests for some other LCM branch, or is this pull-request the first time these changes are being reviewed after backport/cherrypick?

Hui Zhang's commit "CP-14701: backport Xapi ssl_legacy/TLS1.2 changes" (currently 964b428b8ac25569132c2effdf43b6a7deceabac) should have a more meaningful commit-message.
As it is, it is not easy to review because there is nothing to tell me what the change is meant to achieve or why it was needed.
This commit looks as if it combines several commits from xen-api/trunk, so it should list them and include their commit-messages, or preferably an amalgamated version of the messages: there is no need to mention the rolling-pool-upgrade part of the history.

Message updated to say:
Now I have looked at the end result in that file (`ocaml/xapi/xapi_mgmt_iface.ml`) and it looks fine, so I suppose that commit must have got things right. A more meaningful commit-message would still be worthwhile though.

**Comment by thomassa** [comment]
I'm happy with this apart from my comments above.

**Comment by fillzero** [comment]
Hi @thomassa ,
I updated the commits "CP-14701: Backport Xapi ssl_legacy/TLS1.2 changes".
Thanks

**Pull request merged**